src/screens/Auth/Login.js

```diff
@@ -10,18 +17,29 @@ export default class Login extends Component {
     this.state = {
       email: "",
       password: "",
-      waiting: false
+      submitWaiting: false,
+      amazonWaiting: false
     }
 
     this.submit = this.submit.bind(this)
     this.createAccount = this.createAccount.bind(this)
     this.forgotPassword = this.forgotPassword.bind(this)
+    this.loginWithAmazon = this.loginWithAmazon.bind(this)
+  }
+
+  loginWithAmazon() {
+    this.setState({amazonWaiting: true})
+    this.props.amazonLogin().then(() => {
+      this.setState({amazonWaiting: false})
+      if(this.props.User)
```

Comment:
maybe catch this instead

Reply:
Ian made a custom React Native API for performing the login functionality which can be ported via npm which is quite impressive!